pkg/mesos/client.go

```diff
@@ -79,14 +99,32 @@ func (c *Client) StateSummary() (*StateSummary, error) {
 	}
 }
 
-// Slaves returns the agents of the mesos cluster.
-func (c *Client) Slaves() ([]Slave, error) {
-	state, err := c.StateSummary()
+// Agents returns the agents of the mesos cluster.
+func (c *Client) Agents() ([]master.Response_GetAgents_Agent, error) {
```

Comment:
Currently we are only returning types defined in this package (`Master`, `State`, `StateSummary`, etc), thus the mesos-go dependency is not exposed and remains an implementation detailed. Can we follow the same approach here for consistency?

Reply:
(Maybe we should do it the other way around, and update other methods to return "pure" definitions from the structs in `mesos-go`, we just need to decide on which approach to choose and stick to it.)

Reply:
@rgo3 : FYI we're going to look at this separately from this PR.